decompiler/pipeline/controlflowanalysis/restructuring_commons/ast_processor.py

```diff
@@ -108,6 +107,35 @@ def _partition_conditional_breaks_in_groups_and_combine(self, seq_node: SeqNode,
             + last_break_condition
         )
 
+    def _remove_break_condition_from_nodes_behind(self):
+        """Remove break-conditions from the following conditions."""
```

Comment:
I needed a moment to understand this function again. I updated the description and name. 
The idea/problem is the following. We can have the following:
![no_break_0](https://github.com/fkie-cad/dewolf/assets/7224664/506e8024-0b39-48c8-8f0d-2bf48dae26bb)

If the break-condition is fulfilled, then we break and the following will not be evaluated. So, we can assume that the negation of this condition holds at all the nodes after the break-node. Thus, when reaching the continue-node, we know that `arg1 != 5 or var_0 !=5`. If we know this, the condition before the continue-node evaluates to true.

Reply:
And `is_equal` is not expensive, this can be done in linear time, the expensive check is the `is_equivalent`. 
The `is_equal` should be enough here in order to not make it too expensive.